source/tutorials/file-sets.md

```diff
@@ -435,49 +545,72 @@ And then create a file set from just the ones we're interested in:
       ./src
     )
   ];
+in
+
+fs.trace sourceFiles
+
+stdenv.mkDerivation {
+  name = "fileset";
+  src = fs.toSource {
+    root = ./.;
+    fileset = sourceFiles;
+  };
+  postInstall = ''
+    cp -vr $src $out
+  '';
+}
 ```
 
-When building this you'll see that only the specified files are used, even when a new one is added:
+The `postInstall` script is simplified to rely on the sources to be pre-filtered appropriately:
 
 ```shell-session
 $ nix-build
-trace: /home/user/select
+trace: /home/user/fileset
 trace: - build.sh (regular)
-trace: - src
-trace:   - select.c (regular)
-trace:   - select.h (regular)
-trace: - string.txt (regular)
+trace: - hello.txt (regular)
+trace: - src (all files in directory)
+trace: - world.txt (regular)
 this derivation will be built:
-  /nix/store/gzj9j9dk2qyd46y1g2wkpkrbc3f2nm5g-filesets.drv
-building '/nix/store/gzj9j9dk2qyd46y1g2wkpkrbc3f2nm5g-filesets.drv'...
-[...]
-/nix/store/sb4g8skwvpwbay5kdpnyhwjglxqzim28-filesets
+  /nix/store/sjzkn07d6a4qfp60p6dc64pzvmmdafff-fileset.drv
+...
+'/nix/store/6k6pv78b1dkdhbdlzxar5vb3z3c8fwza-source' -> '/nix/store/zl4n1g6is4cmsqf02
+dci5b2h5zd0ia4r-fileset'
+'/nix/store/6k6pv78b1dkdhbdlzxar5vb3z3c8fwza-source/build.sh' -> '/nix/store/zl4n1g6i
+s4cmsqf02dci5b2h5zd0ia4r-fileset/build.sh'
+'/nix/store/6k6pv78b1dkdhbdlzxar5vb3z3c8fwza-source/hello.txt' -> '/nix/store/zl4n1g6
+is4cmsqf02dci5b2h5zd0ia4r-fileset/hello.txt'
+'/nix/store/6k6pv78b1dkdhbdlzxar5vb3z3c8fwza-source/world.txt' -> '/nix/store/zl4n1g6
+is4cmsqf02dci5b2h5zd0ia4r-fileset/world.txt'
+'/nix/store/6k6pv78b1dkdhbdlzxar5vb3z3c8fwza-source/src' -> '/nix/store/zl4n1g6is4cms
+qf02dci5b2h5zd0ia4r-fileset/src'
+'/nix/store/6k6pv78b1dkdhbdlzxar5vb3z3c8fwza-source/src/select.c' -> '/nix/store/zl4n
+1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/src/select.c'
+'/nix/store/6k6pv78b1dkdhbdlzxar5vb3z3c8fwza-source/src/select.h' -> '/nix/store/zl4n
+1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/src/select.h'
```

Comment:
```suggestion
'.' -> '/nix/store/zl4n1g6is4cmsqf02dci5b2h5zd0ia4r-fileset'
'./build.sh' -> '/nix/store/zl4n1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/build.sh'
'./hello.txt' -> '/nix/store/zl4n1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/hello.txt'
'./world.txt' -> '/nix/store/zl4n1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/world.txt'
'./src' -> '/nix/store/zl4n1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/src'
'./src/select.c' -> '/nix/store/zl4n1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/src/select.c'
'./src/select.h' -> '/nix/store/zl4n1g6is4cmsqf02dci5b2h5zd0ia4r-fileset/src/select.h'
```

With the `cp` change

Reply:
Eh. Okay, that's actually a good argument to do that.